### PR TITLE
Enable additional `maven-enforcer-plugin` rules

### PR DESCRIPTION
**Pull request by Stephan202**
Suggested commit message:
```
Enable additional `maven-enforcer-plugin` rules (#489)
```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -904,8 +904,37 @@
...
+                                            <ignoreClass>org.checkerframework.dataflow.qual.*</ignoreClass>
+                                        </ignoreClasses>
+                                    </dependency>
+                                </dependencies>
+                                <findAllDuplicates>true</findAllDuplicates>
+                            </banDuplicateClasses>
                             <banDuplicatePomDependencyVersions />
+                            <!-- XXX: Enable this rule once it no longer
+                            "downloads the internet". See
+                            https://issues.apache.org/jira/browse/MENFORCER-467.
```
Comment (Stephan202) [inline pom.xml:932]: [Link](https://issues.apache.org/jira/browse/MENFORCER-467) for easy clicking.

File: pom.xml [hunk 2]
```diff
@@ -904,8 +904,37 @@
...
+                            <banDynamicVersions />-->
                             <dependencyConvergence />
+                            <enforceBytecodeVersion>
```
Comment (rickie) [inline pom.xml:935]: Another [link](https://www.mojohaus.org/extra-enforcer-rules/enforceBytecodeVersion.html) for easy clicking.

**Review by rickie — APPROVED** [review]
Changes LGTM!

**Review by chamil-prabodha — APPROVED** [review]
Some of the rules are documented here and might be useful to link it here :)
https://www.mojohaus.org/extra-enforcer-rules/

**Pull request merged**